src/pages/SetPasswordPage.js

```diff
@@ -69,14 +70,26 @@ class SetPasswordPage extends Component {
      * Validate the form and then submit it
      */
     validateAndSubmitForm() {
+        const accountID = lodashGet(this.props.route.params, 'accountID', '');
+        const validateCode = lodashGet(this.props.route.params, 'validateCode', '');
         if (!this.state.isFormValid) {
             return;
         }
-        setPassword(
-            this.state.password,
-            lodashGet(this.props.route, 'params.validateCode', ''),
-            lodashGet(this.props.route, 'params.accountID', ''),
-        );
+        API.ValidateEmail({
+            accountID,
+            validateCode,
+        }).then((responseValidate) => {
+            if (responseValidate.jsonCode === 200) {
+                API.ChangePassword({
+                    authToken: responseValidate.authToken,
+                    password: this.state.password,
+                }).then((responsePassword) => {
+                    if (responsePassword.jsonCode === 200) {
+                        signIn(this.state.password);
```

Comment:
Why do we need to call signIn? Is it because validate and changePassword do not set the authToken cookie or what?

Reply:
Yep. SignIn sets the cookie and redirects us to where we want to go, I believe

Reply:
I don't think there are any cookies in New Expensify. We mostly need to do this to create the "device login" to make the re-authentication logic work correctly.